### DISCUSSION TITLE
Incorrect merged/concatenated files duration

### DISCUSSION THREAD

**Discussion by Infinitay** [mifi/lossless-cut#1253] [Support] [answered]
I know there are some issues with duration when it comes to cutting and merging the segments; however, what about when merging multiple files together found under the tools option? For example a 24 minute video and a 39 minute video merges into a 2 hour video for me.

I should also note that one file is a MPEG-4 and the other is a MPEG-TS. I didn't notice this at first, but would this be the problem as to why?
![image](https://user-images.githubusercontent.com/6964154/182027752-8ba1fbba-7070-4907-9ee0-d30573edab22.png)

<details>
<summary>File A Details</summary>

```
General
Complete name               : A.mp4
Format                      : MPEG-4
Format profile              : Base Media
Codec ID                    : isom (isom/iso2/avc1/mp41)
File size                   : 699 MiB
Duration                    : 24 min 58 s
Overall bit rate            : 3 913 kb/s
Writing application         : Lavf59.16.100

Video
ID                          : 1
Format                      : AVC
Format/Info                 : Advanced Video Codec
Format profile              : High@L3.1
Format settings             : CABAC / 1 Ref Frames
Format settings, CABAC      : Yes
Format settings, Reference  : 1 frame
Format settings, GOP        : M=1, N=15
Codec ID                    : avc1
Codec ID/Info               : Advanced Video Coding
Duration                    : 24 min 58 s
Bit rate                    : 3 775 kb/s
Width                       : 1 280 pixels
Height                      : 720 pixels
Display aspect ratio        : 16:9
Frame rate mode             : Variable
Frame rate                  : 29.926 FPS
Minimum frame rate          : 0.636 FPS
Maximum frame rate          : 30.303 FPS
Color space                 : YUV
Chroma subsampling          : 4:2:0
Bit depth                   : 8 bits
Scan type                   : Progressive
Bits/(Pixel*Frame)          : 0.137
Stream size                 : 674 MiB (96%)
Codec configuration box     : avcC

Audio
ID                          : 2
Format                      : AAC LC
Format/Info                 : Advanced Audio Codec Low Complexity
Codec ID                    : mp4a-40-2
Duration                    : 24 min 58 s
Source duration             : 24 min 58 s
Bit rate mode               : Constant
Bit rate                    : 130 kb/s
Channel(s)                  : 2 channels
Channel layout              : L R
Sampling rate               : 48.0 kHz
Frame rate                  : 46.875 FPS (1024 SPF)
Compression mode            : Lossy
Stream size                 : 23.2 MiB (3%)
Source stream size          : 23.2 MiB (3%)
Default                     : Yes
Alternate group             : 1
```

</details>

<details>
<summary>File B Details</summary>

```
General
ID                          : 1 (0x1)
Complete name               : B.mp4
Format                      : MPEG-TS
File size                   : 1.14 GiB
Duration                    : 39 min 22 s
Overall bit rate mode       : Variable
Overall bit rate            : 4 129 kb/s
FileExtension_Invalid       : ts m2t m2s m4t m4s tmf ts tp trp ty

Video
ID                          : 256 (0x100)
Menu ID                     : 1 (0x1)
Format                      : AVC
Format/Info                 : Advanced Video Codec
Format profile              : Main@L4
Format settings             : CABAC / 4 Ref Frames
Format settings, CABAC      : Yes
Format settings, Reference  : 4 frames
Codec ID                    : 27
Duration                    : 39 min 22 s
Width                       : 1 920 pixels
Height                      : 1 080 pixels
Display aspect ratio        : 16:9
Frame rate mode             : Variable
Color space                 : YUV
Chroma subsampling          : 4:2:0
Bit depth                   : 8 bits
Scan type                   : Progressive
Color range                 : Limited
Color primaries             : BT.709

Audio
ID                          : 257 (0x101)
Menu ID                     : 1 (0x1)
Format                      : AAC LC
Format/Info                 : Advanced Audio Codec Low Complexity
Format version              : Version 4
Muxing mode                 : ADTS
Codec ID                    : 15-2
Duration                    : 39 min 22 s
Bit rate mode               : Variable
Channel(s)                  : 2 channels
Channel layout              : L R
Sampling rate               : 44.1 kHz
Frame rate                  : 43.066 FPS (1024 SPF)
Compression mode            : Lossy

Menu
ID                          : 4096 (0x1000)
Menu ID                     : 1 (0x1)
Format                      : AVC / AAC
Duration                    : 39 min 22 s
List                        : 256 (0x100) (AVC) / 257 (0x101) (AAC)
Service name                : Service01
Service provider            : FFmpeg
Service type                : digital television
```

</details>

**Comment by mifi** [comment, marked as answer]
hi. your files have different codec parameters, so i think that's probably why you're having trouble. (this is not supported)

**Reply by Infinitay** [reply]
That was it. I didn't think it would be an issue because from my little understanding I assumed MPEG-4 and MPEG-TS wouldn't have issues with each other.

Would you consider a new feature that notifies the user of differing codecs and to prompt them if they would first like to convert all files to a certain codec prior to merging appropriate?

**Reply by hrxn** [reply]
This is not really because of MPEG-4 and MPEG-TS, as listed here in your example. This alone would work.
First of all, these are not the codecs, these are the (file) formats, also known as containers (container formats).

MPEG-4 itself is the name of an entire standard, but what's usually meant here (including your example) is what is colloquially simply called an MP4 File, which usually has the `.mp4`  extension, which is an container format derived from `.mov` and is actually called ISOBMFF (ISO base media file format (aka MPEG-4 Part 12)).

MPEG-TS on the other hand is the MPEG transport stream format, and while they both share the MPEG name, and they are both considered container formats, they are actually very different.

But the problem in your case are the video codecs.
As shown in your MediaInfo output, they are both AVC, but that's where the similarities already stop. They have different base profiles and levels ("Main" vs. "High", etc.) and different codec parameters, as has been pointed out by mifi. They even have different image dimensions (1080p vs. 720p).

So, there's no way around to transcode one of the two streams to match the other stream. There are different approaches here, depending on the desired output result you aim for..

**Reply by mifi** [reply]
I don't know which version you're running @Infinitay - but the newest v3.46.2 will give a warning if trying to concat files with different codec parameters. I'm planning on releasing this to the Stores soon as well.

**Reply by Infinitay** [reply]
Well that's embarrassing. I didn't even notice the video dimensions differed from one another. Thank you for your insightful post @HRXN.

Nice to hear that the upcoming version will offer better _support_ for this. Thank you.
